library/std/src/sys/dbg.rs

```diff
@@ -0,0 +1,216 @@
+//! Debugging aids.
+
+/// Presence of a debugger.
+#[derive(Copy, Clone, Debug)]
+pub(crate) enum DebuggerPresence {
+    /// The debugger is attached to the process.
+    Detected,
+    /// The debugger is not attached to the process.
+    NotDetected,
+}
+
+/// Error detecting debugger presence.
+#[derive(Copy, Clone, Debug)]
+pub(crate) enum DebuggerPresenceError {
+    /// The functionality is not available.
+    NotImplemented,
+    /// The detection logic failed to determine
+    /// the debugger presence. It may or may not be
+    /// attached.
+    #[cfg(any(target_os = "macos", target_os = "freebsd", target_os = "linux"))]
+    DetectionFailed,
+}
```

Comment:
You can replace this enum with converting the relevant `Result` to an `Option` without loss of meaning, and that avoids the awkward `cfg`.

Reply:
Very satisfying now, thank you!